core/capabilities/streams/trigger_test.go

```diff
@@ -0,0 +1,224 @@
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/chaintype"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/ocr2key"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+)
+
+const (
+	triggerID          = "streams-trigger@1.0.0"
+	workflowID         = "workflowID1"
+	baseTriggerEventID = "triggerEvent_"
+	basePrice          = 2000000000
+	baseTimestamp      = 1000000000
+)
+
+type feed struct {
+	feedID    [32]byte
+	feedIDStr string
+	reports   []report
+}
+
+type report struct {
+	rawReport  []byte
+	reportCtx  []byte
+	signatures [][]byte
+}
+
+type node struct {
+	peerID p2ptypes.PeerID
+	bundle ocr2key.KeyBundle
+}
+
+// Integration/load test that combines Trigger Subscriber, Streams Trigger Aggregator and Streams Codec.
+// It measures time needed to receive and process trigger events from multiple nodes and produce a local aggregated event.
+// For more meaningful measurements, increase the values of parameters P and T.
+func TestStreamsTrigger(t *testing.T) {
+	N := 31 // trigger DON nodes
+	F := 10 // faulty nodes
+	R := 5  // different reports per feed (i.e. prices and timestamps)
+	P := 2  // feeds
+	T := 2  // test iterations
+
+	nodes := newNodes(t, N)
+	feeds := newFeedsWithSignedReports(t, nodes, N, P, R)
+
+	allowedSigners := make([][]byte, N)
+	for i := 0; i < N; i++ {
+		allowedSigners[i] = nodes[i].bundle.PublicKey() // bad name - see comment on evmKeyring.PublicKey
+	}
+	lggr := logger.TestLogger(t)
+	ctx := testutils.Context(t)
+	codec := streams.NewCodec(lggr)
+	agg := triggers.NewMercuryRemoteAggregator(codec, allowedSigners, F, lggr)
+
+	capInfo := capabilities.CapabilityInfo{
+		ID: triggerID,
+	}
+	capMembers := make([]p2ptypes.PeerID, N)
+	for i := 0; i < N; i++ {
+		capMembers[i] = nodes[i].peerID
+	}
+	capDonInfo := capabilities.DON{
+		Members: capMembers,
+		F:       uint8(F),
+	}
+	config := &remotetypes.RemoteTriggerConfig{
+		MinResponsesToAggregate: uint32(F + 1),
+	}
+	subscriber := remote.NewTriggerSubscriber(config, capInfo, capDonInfo, capabilities.DON{}, nil, agg, lggr)
+
+	// register trigger
+	req := capabilities.CapabilityRequest{
+		Metadata: capabilities.RequestMetadata{
+			WorkflowID: workflowID,
+		},
+	}
+	triggerEventCallbackCh, err := subscriber.RegisterTrigger(ctx, req)
+	require.NoError(t, err)
+
+	// send and process all trigger events
+	startTs := time.Now().UnixMilli()
+	processingTime := int64(0)
+	for c := 0; c < T; c++ {
+		triggerEventID := baseTriggerEventID + strconv.Itoa(c)
+		for i := 0; i < N; i++ { // every node ...
+			reportList := make([]datastreams.FeedReport, P)
+			for j := 0; j < P; j++ { //  ... sends reports for every feed ...
+				reportIdx := (i + j) % R
+				signatures := make([][]byte, F+1)
+				for k := 0; k < F+1; k++ { // ... each signed by F+1 nodes
```

Comment:
Do we want to profile failure scenarios? Or are we good with just happy path load testing?

Reply:
I think we're good. Failures shouldn't be heavier than successful ones.

Although that makes me think that we might need some extra protections agains malicious nodes trying to attach too many signatures - I might add something in a separate PR, thanks!